force-app/main/default/classes/BDI_MatchDonations.cls

```diff
@@ -430,94 +430,105 @@ public with sharing class BDI_MatchDonations implements BDI_IMatchDonations {
+        List<Opportunity> listOpp = Database.query(buildPotentialDonationMatchingQuery(
+            opportunityIds, accountIds, contactIds));
+        pl.stop();
+        return listOpp;           
+    }
 
-        // need to include any Pmt fields we may update, so we can compare against them
-        setPmtFields.addAll(mapDIFieldToPmtField.values());
+    private String buildPotentialDonationMatchingQuery(Set<Id> opportunityIds, Set<Id> accountIds, Set<Id> contactIds) {
+        Integer cFilter = 0;
+        String strSoql = 'SELECT ' + String.join(new List<String>(opportunityFields()), ',');
 
-        // now query for all Open Opportunities from these Contacts or Accounts
-        integer cFilter = 0;
-        string strSoql = 'SELECT ' + string.join(new list<String>(setOppFields), ',');
-        
         // add Payment subquery
-        strSoql += ', (SELECT ' + string.join(new list<String>(setPmtFields), ',');
+        strSoql += ', (SELECT ' + String.join(new List<String>(paymentFields()), ',');
         strSoql += ' FROM npe01__OppPayment__r ' +
-                'WHERE npe01__Paid__c = false ' +
-                'AND npe01__Written_Off__c = false)';
+            'WHERE npe01__Paid__c = false ' +
+            'AND npe01__Written_Off__c = false)';
 
         strSoql += ' FROM Opportunity';
         strSoql += ' WHERE IsClosed = false AND (';
-        if (setOppId.size() > 0) {
+        if (opportunityIds.size() > 0) {
             strSoql += ' Id IN :setOppId ';
             cFilter++;
         }
-        if (setAccId.size() > 0) {
+        if (accountIds.size() > 0) {
             if (cFilter > 0) {
                 strSoql += ' or ';
             }
             strSoql += ' AccountId IN :setAccId ';
             cFilter++;
         }
-        if (setConId.size() > 0) {
+        if (contactIds.size() > 0) {
             if (cFilter > 0) {
                 strSoql += ' OR ';
             }
             strSoql += ' ' + UTIL_Namespace.StrTokenNSPrefix('Primary_Contact__c') + ' IN :setConId ';
             cFilter++;
         }
-        strSoql += ') ';
-        BDI_PerfLogger.PerfLog pl = BDI_DataImportService.perfLogger.newPerfLog('SOQL existing Opps', BDI_PerfLogger.QUERY);
-        list<Opportunity> listOpp = database.query(strSoql);
-        pl.stop();
-        return listOpp;           
+        strSoql += ') LIMIT 50000';
+        return strSoql;
+    }
+
+    private void populateQueryFilterSetsFrom(
+        List<DataImport__c> dataImports,
+        Set<Id> setOppId,
+        Set<Id> setConId,
+        Set<Id> setAccId) {
+        for (DataImport__c di : dataImports) {
+            Contact con = bdi.contactFromDi(di, 1);
+            Account acc = bdi.AccountFromDi(di, 1);
+            // handle already been given the match!
+            if (di.DonationImported__c != null) {
+                setOppId.add(di.DonationImported__c);
+            } else if (di.Donation_Donor__c == 'Contact1' && con.Id != null) {
```

Comment:
Hey @callen-sfdo and @pedrodalcol, I just made this change to prevent the query built for potential donation matching from filtering on null account or contact ids on dry-run through BDI. I wanted to put it on your team's radar so you are aware of this change. This change fixes this issue #6580. Open to all feedback!

Reply:
@voduyemi Hi Victor, thanks for looping us in on this, I just reviewed the change and I don't think it should cause any problems for us.  We actually skip most of the donation matching logic for elevate payment related DIs anyway.  cc @pedrodalcol